clang/lib/Serialization/GeneratePCH.cpp

```diff
@@ -101,12 +102,26 @@ ReducedBMIGenerator::ReducedBMIGenerator(const Preprocessor &PP,
 
 Module *ReducedBMIGenerator::getEmittingModule(ASTContext &Ctx) {
   Module *M = Ctx.getCurrentNamedModule();
-  assert(M->isNamedModuleUnit() &&
+  assert(M && M->isNamedModuleUnit() &&
          "ReducedBMIGenerator should only be used with C++20 Named modules.");
   return M;
 }
 
 void ReducedBMIGenerator::HandleTranslationUnit(ASTContext &Ctx) {
+  // FIMXE: We'd better to wrap such options to a new class ASTWriterOptions.
+  getPreprocessor()
```

Comment:
What's the motivation for this change/how does it relate to the purpose of this patch?

(& it might be easier to read if the HeaderSearchOpts were accessed once, stored in a reference, then the various settings were made:
```
HeaderSearchOptions& Opts = getPreprocessor().getHeaderSearchInfo().getHeaderSearchOpts();
Opts.ModulesSkipDiagnosticOptions = true;
Opts.ModulesSkipHeaderSearchPaths = true;
Opts.ModulesSkipPragmaDiagnosticMappings = true;
```
)

Reply:
Oh, my bad. The motivation is that during my local testing, I found the size of the reduced BMI may be larger than the full BMI, which should be incorrect. And then I tried to fix this on the fly... I've splitted this into a NFC patch https://github.com/llvm/llvm-project/commit/ed1cfffe9b2b2d3cc9279ff83400ace156b317a2